pkg/chunkenc/pool.go

```diff
@@ -28,7 +28,7 @@ var (
 		},
 	}
 	// BytesBufferPool is a bytes buffer used for lines decompressed.
-	BytesBufferPool = newBufferPoolWithSize(4096)
+	BytesBufferPool = pool.New(1024, 16384, 2, func(size int) interface{} { return make([]byte, 0, size) })
```

Comment:
where did we get 1024 and 16384 from? I understand it's 2^10 and 2^14 but they seem a little magic number-y

Reply:
Buckets created will be: `[1KB, 2KB, 4KB, 8KB, 16KB]` now may be the 16KB is too much to keep ? WDYT

Reply:
I think most line will be below 1KB anyways so I'm even wondering if we should not get rid of 8KB too.

The idea is, if the buffer is too small we will swap it with a higher bucket or the exact value if the bucket doesn't exist, each block iterator keeps a buffer for their line, and release it once the end of a block is reached.

Reply:
Gotcha, I'll leave the bucket sizes up to your judgement,